### ISSUE TITLE
Trailing comma after function return shorthand produce SyntaxError output

### ISSUE THREAD

**Issue by meriadec** [prettier/prettier#3173]
**Prettier 1.8.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgCU4BDMAgMxwnXwHIdTy6AdKNuAD2z3wBM4FEgFcANgQAUwfNABiECPgC+ASnwBeAHz4JbfPgA8fVADc9+mVADCo1GADW64BLVbzFy-IhT3H-ACMSHCR6QIAvOgAaX31VXyVzAHpNNhUQSJAITBhUaABnZFBMHFRYAHVUPhgAC2QADgAGDJgSfwqq2qQAJgy80oBzUTgARWEIeGQhUTy4ZpwSVFsofqsadBJkEBJRUXSQf3mHOBgAZUwyAeQYHGFZkAArPM4AIUP7Y5OSdDgAGVK4SbbGYZc44GY4TaBfwAT1E0D2M3QqCuNzuwhmABVWgUkFNgSA+BAwIDpncSHkYCT8RBhDBMLSulS4EolEA)
```sh
--trailing-comma all
```

**Input:**
```jsx
import React from 'react'

export default ({ onFoo }) => (
  <div
    onClick={() =>
      onFoo({
        bar: 'baz',
      })
    }
  />
)
```

**Output:**
```jsx
import React from "react";

export default ({ onFoo }) => (
  <div
    onClick={() =>
      onFoo({
        bar: "baz",
      }),
    }
  />
);

```

**Expected behavior:**

There should not be a trailing comma after `onFoo(...)`. This is causing an error when running prettier a second time on the file.

Thank you for your time, and for this project!

**Comment by azz** [comment]
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgCU4BDMAgMxwnXwHIdTy6AdKNuAD2z3wBM4FEgFcANgQAUwfNABiECPgC+ASnwBeAHz4JbfPgA8fVADc9+mVADCo1GADW64BLVbzFy-IhT3H-ACMSHCR6QIAvOgAaX31VXyVzAHpNNhUQSJAITBhUaABnZFASPJhkIVE8uAz-HDJ7OBgAZUwyVCgAc2QYHGEqkD4IMDKSCr6AKzzOACFahwbGknQ4ABk2uGHRjIhhGEwdgCYunr6WnEqcZBBA-wBPUWh0kEwcNpgAdVQ+GAALZAAOAAMGUq6FQR16wLa7VEcAAisIIPANpUMjASP4Pl9fkh9qjaqhbB0rDR0CRLiNRI9hJUACrogpIcqVJRKIA)


**Second Output:**
```jsx
SyntaxError: Unexpected token (9:5)
   7 |         bar: "baz",
   8 |       }),
>  9 |     }
     |     ^
  10 |   />
  11 | );
  12 | 
```

**Comment by azz** [comment]
This has been fixed in `1.8.1`.
